### PR TITLE
Set up semantic release

### PR DESCRIPTION
**Pull request by andrii-balitskyi**
Closes https://github.com/seamapi/seam-cli/issues/10

### CODE REVIEW

File: .github/workflows/semantic-release.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+---
+name: Semantic Release
+
+run-name: Semantic Release from ${{ github.ref_name }}
+
+on:
+  push:
+    branches:
+      # - "**"
+      - "TODO: Enable once version 1.0.0 is released."
```
Comment (seveibar) [inline .github/workflows/semantic-release.yml:10]: can we use the semantic release system from [makenew-tsmodule](https://github.com/seamapi/makenew-tsmodule)? I'm a bit confused by some of the things here e.g. the TODO

**Review by seveibar — COMMENTED** [review]
i think we should use the more standard release style

File: .github/workflows/semantic-release.yml (hunk 1 above)
Comment (andrii-balitskyi) [reply .github/workflows/semantic-release.yml]: https://github.com/seamapi/seam-cli/pull/14/commits/f87563308d2e7834e67bedb4e140e2ddc2333621

File: .releaserc.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+  "branches": [
+    "+([0-9])?(.{+([0-9]),x}).x",
+    "main",
+    "next",
+    "next-major",
+    { "name": "beta", "prerelease": true },
+    { "name": "alpha", "prerelease": true }
+  ],
+  "plugins": ["@semantic-release/commit-analyzer"]
```
Comment (seveibar) [inline .releaserc.json:10]: do these need to be installed in package.json

Comment (andrii-balitskyi) [reply .releaserc.json:10]: I figured that they don't as they are not in the [package.json](https://github.com/seamapi/makenew-tsmodule/blob/2887311c5d6d1b660dc281a57c526c173e88de12/package.json#L60) on makenew-tsmodule

**Review by seveibar — APPROVED** [review]

File: .github/actions/setup/action.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    description: The Node.js auth token.
+    required: true
+  node_version:
+    description: The Node.js version.
+    required: false
+    default: "20"
+  registry_url:
+    description: The Node.js package registry URL.
+    required: false
+    default: https://npm.pkg.github.com
```
Comment (seveibar) [inline .github/actions/setup/action.yml:16]: @andrii-balitskyi this line should be omitted for public packages

**Review by seveibar — APPROVED** [review]

**Comment by seveibar** [comment]
shouldn't be needed if https://github.com/seamapi/seam-cli/pull/72 works!

**Comment by andrii-balitskyi** [comment]
Closing this to introduce Seve's new versioning CLI `pver`

**Pull request closed without merging**